x/evm/keeper/state_transition.go

```diff
@@ -175,6 +175,9 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 		panic("context stack shouldn't be dirty before apply message")
 	}
 
+	// Contains the tx processing and post processing in same scope
+	revision := k.Snapshot()
```

Comment:
I thought this was done internally at during the EVM call/create, or if we do that it doesn't include the post-processing?

Can you update the ADR and add this line?

Reply:
> I thought this was done internally at during the EVM call/create, or if we do that it doesn't include the post-processing?

Yes, we need this to include the post-processing step.

> 
> Can you update the ADR and add this line?

sure

Reply:
done